research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,289 @@
+import os
+import sys
+import time
+import shutil
+import zipfile
+import urllib.request
+
+import pandas as pd
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+DATA_PATH = os.path.join(BASE_DIR, 'data')
+
+
+def reporthook(blocknum, blocksize, totalsize):
+    readsofar = blocknum * blocksize
+    if totalsize > 0:
+        percent = readsofar * 1e2 / totalsize
+        s = "\r%5.1f%% %*d / %d" % (
+            percent, len(str(totalsize)), readsofar, totalsize)
+        sys.stderr.write(s)
+        if readsofar >= totalsize:
+            sys.stderr.write("\n")
+    else:
+        sys.stderr.write("read %d\n" % (readsofar,))
+
+
+def folder_walk(year):
+    ret_dict = {}
+    if year == '2010':
+        donations_data_candidates = []
+        donations_data_parties = []
+        donations_data_committees = []
+        for root, dirs, files in os.walk("prestacao_contas_2010", topdown=False):
+            for name in files:
+                if 'Receitas' in name:
+                    data = pd.read_csv(os.path.join(root, name), low_memory=False,
+                                       encoding="ISO-8859-1", sep=';')
+                    if 'candidato' in os.path.join(root, name):
+                        donations_data_candidates.append(data)
+                    elif 'comite' in os.path.join(root, name):
+                        donations_data_committees.append(data)
+                    elif 'partido' in os.path.join(root, name):
+                        donations_data_parties.append(data)
+
+        donations_data_candidates = pd.concat(donations_data_candidates)
+        donations_data_parties = pd.concat(donations_data_parties)
+        donations_data_committees = pd.concat(donations_data_committees)
+        shutil.rmtree('prestacao_contas_2010')
+        ret_dict = {'candidates': donations_data_candidates,
+                    'parties': donations_data_parties,
+                    'committees': donations_data_committees}
+    else:
+        if year == '2012':
+            path_candid = os.path.join('prestacao_final_2012',
+                                       'receitas_candidatos_2012_brasil.txt')
+            path_parties = os.path.join('prestacao_final_2012',
+                                        'receitas_partidos_2012_brasil.txt')
+            path_committ = os.path.join('prestacao_final_2012',
+                                        'receitas_comites_2012_brasil.txt')
+        elif year == '2014':
```

Comment:
Hi @lacerdamarcelo, here what you think about changing for `2012` and `2014`, and then you can do something like: 
```console
path_candid = os.path.join('prestacao_final_{}'.format(year),
                                       'receitas_candidatos_{}_brasil.txt'.format(year))
```